### PR TITLE
Fancy Javascript Example

### PR DESCRIPTION
**Pull request by jbolda**
There is still stuff I could add (especially to the README), but wanted to get this out before I had to jump back to paid work. I tried to comment the files themselves well at least.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit b97801d19ba9cc9bf249c134de044d5ced106428

https://app.netlify.com/sites/using-styled-components/deploys/5966e30e8ebdd909fb189297

File: examples/fancy-javascript/package.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/gatsbyjs/gatsby/issues"
+  },
+  "homepage": "https://www.gatsbyjs.org",
+  "dependencies": {
+    "bulma": "0.4.2",
+    "d3": "4.9.1",
+    "gatsby": "^1.1.0",
+    "gatsby-link": "^1.1.0",
```
Comment (KyleAMathews) [inline examples/fancy-javascript/package.json]: Many of these plugins don't have a 1.1.0 release yet — Lerna doesn't make a new release for a package unless there's been changes.

File: examples/fancy-javascript/.babelrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ['react', 'es2015', 'stage-0'],
```
Comment (jbolda) [inline examples/fancy-javascript/.babelrc]: Gatsbygram had them so I figured I would include them. Can remove them though.

File: examples/fancy-javascript/package.json (hunk 1 above)
Comment (jbolda) [inline examples/fancy-javascript/package.json]: Haha, I was wondering, yarn makes it to easy to ignore this fact.

**Comment by jbolda** [comment]
So my expectation with this example is to showcase any plugin/transformer/etc. that deals directly with javascript. For now, it is just frontmatter, yes. But as we add more js stuff, I would presume that it will be a little bit more complicated then say, markdown, to compose and it feels like it would be nice have an example documenting how they interact / work together. (I mostly showcased the markdown as an addition to this as it is painful to write directly in javascript, and the convenience of markdown I presume will make it a very typical pair.)

That being said, I am not opposed to a name change, but I think it would be worthwhile to keep it a little bit more... general I guess? What are your thoughts?

**Comment by KyleAMathews** [comment]
So right now we have three types of example sites — full-blown applications like gatsbygram, very small plugin example sites, and one "ecosystem" example site https://using-remark.gatsbyjs.org/

I can see this site also becoming a kind of "ecosystem" site. If that's your goal, perhaps name the site "using-javascript-transforms"?

**Comment by jbolda** [comment]
Yea, definitely, that sounds perfect to me!

**Comment by jbolda** [comment]
I think that covers everything.

**Comment by KyleAMathews** [comment]
Awesome! Thanks so much Jacob!

**Comment by KyleAMathews** [comment]
Hey just tried building this on Netlify and got this error:

```
12:50:29 PM: Error: ReferenceError: location is not defined
at SiteSidebar.render (render-page.js:64188:21)
```

Could you look into it and PR a fix?

**Comment by jbolda** [comment]
Will do!

**Pull request merged**